src/main/java/hudson/plugins/analysis/graph/CategoryBuildResultGraph.java

```diff
@@ -234,7 +237,18 @@ protected JFreeChart createChart(final GraphConfiguration configuration, final R
         }
         return valuesPerBuild;
     }
-
+    private boolean passesFilteringByParameter(final AbstractBuild<?, ?> build, final String parameterName, final String parameterValue){
+        if (parameterName == null){
+            return true;
+        }
+        Map<String, String> vars = build.getBuildVariables();
+        if(vars.containsKey(parameterName)){
+            if (vars.get(parameterName).equals(parameterValue)) {
```

Comment:
No: vars.get(parameterName)
It this is null then we get a NPE. 
Can you check what happens when the parameter or value is not set in the UI. Is it the empty string or null?